### PR TITLE
fix some clippy warnings in tests

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

**Comment by cakebaker** [comment]
How do you run clippy? With `cargo clippy --all-targets` I get a single warning in `test_du.rs`

**Comment by sylvestre** [comment]
with clippy nightly:
 cargo clippy --fix -- -W clippy::pedantic

File: tests/by-util/test_tee.rs [hunk 1]
```diff
@@ -118,9 +118,10 @@ mod linux_only {
         use std::os::unix::io::FromRawFd;
 
         let mut fds: [c_int; 2] = [0, 0];
-        if unsafe { libc::pipe(&mut fds as *mut c_int) } != 0 {
-            panic!("Failed to create pipe");
-        }
+        assert!(
+            !(unsafe { libc::pipe(&mut fds as *mut c_int) } != 0),
```
Comment (cakebaker) [inline tests/by-util/test_tee.rs:122]:
```suggestion
            (unsafe { libc::pipe(&mut fds as *mut c_int) } == 0),
```

File: tests/by-util/test_cp.rs [hunk 2]
```diff
@@ -1392,15 +1392,14 @@ fn test_cp_target_file_dev_null() {
 fn test_cp_one_file_system() {
     use crate::common::util::AtPath;
     use walkdir::WalkDir;
-
```
Comment (cakebaker) [inline tests/by-util/test_cp.rs:1395]: Why do you remove this line? Seems like a useful separator between imports and variable declarations.

File: tests/by-util/test_cp.rs [hunk 3]
```diff
@@ -1392,15 +1392,14 @@ fn test_cp_target_file_dev_null() {
...
     let scene = TestScenario::new(util_name!());
```
Comment (sylvestre) [inline tests/by-util/test_cp.rs:1395]:
```suggestion

    let scene = TestScenario::new(util_name!());
```

**Pull request merged**